build_js.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+git submodule update --init --recursive
+
+DATA_DIR="$(pwd)/data/"
+
+mkdir -p build_js
+cd build_js
+
+cmake ../src \
+    -DBUILD_WEBGL=ON \
+    -DBUILD_GUI_VIEWERS=ON \
+    -DBUILD_PYTHON_BINDINGS=OFF \
+    -DBUILD_DATATOOL=OFF \
+    -DBUILD_PTEX_SUPPORT=OFF \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_PREFIX_PATH="$EMSCRIPTEN" \
+    -DTARGET_GLES2=OFF \
+    -DTARGET_WEBGL=ON \
+    -DCMAKE_TOOLCHAIN_FILE="../src/deps/corrade/toolchains/generic/Emscripten-wasm.cmake" \
+    -DCMAKE_INSTALL_PREFIX="." \
+    -DCMAKE_CXX_FLAGS="-s FORCE_FILESYSTEM=1 -s TOTAL_MEMORY=1677721600 --preload-file $DATA_DIR/scene_datasets/habitat-test-scenes@/" \
```

Comment:
I think [`-s ALLOW_MEMORY_GROWTH=1`](https://emscripten.org/docs/optimizing/Optimizing-Code.html#memory-growth) would be a better option here, it has no perf penalty on Wasm.

Just FYI -- unfortunately as far as I checked the `--preload-file` bundles the files as base64 inside `*.js` instead of (the obvious) binary inside `*.wasm`, which makes the download sizes *extra bad* :(

Reply:
Good to know that `ALLOW_MEMORY_GROWTH=1` has no perf penalty. Packaging the data with `--preload-file` is a temporary kludge that I used for testing. We can't rely on packaging all of our assets in this way.  A filesystem abstraction transparently supporting web assets is the proper solution to this 😅.